tests/e2e/appworker_test.go

```diff
@@ -0,0 +1,173 @@
+		Setup(func(ctx context.Context, tt *testing.T, cfg *envconf.Config) context.Context {
+			namespace := GetNamespaceForFeature(ctx)
+
+			client, err := cfg.NewClient()
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// create the postgres
+			err = deployAndWaitForPostgres(ctx, cfg, namespace)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			connectAddr := fmt.Sprintf("postgres.%s:5432", namespace)
+
+			// create the temporal cluster
+			cluster = &v1beta1.TemporalCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: namespace,
+				},
+				Spec: v1beta1.TemporalClusterSpec{
+					NumHistoryShards:           1,
+					JobTtlSecondsAfterFinished: &jobTtl,
+					Persistence: v1beta1.TemporalPersistenceSpec{
+						DefaultStore: &v1beta1.DatastoreSpec{
+							SQL: &v1beta1.SQLSpec{
+								User:            "temporal",
+								PluginName:      "postgres",
+								DatabaseName:    "temporal",
+								ConnectAddr:     connectAddr,
+								ConnectProtocol: "tcp",
+							},
+							PasswordSecretRef: v1beta1.SecretKeyReference{
+								Name: "postgres-password",
+								Key:  "PASSWORD",
+							},
+						},
+						VisibilityStore: &v1beta1.DatastoreSpec{
+							SQL: &v1beta1.SQLSpec{
+								User:            "temporal",
+								PluginName:      "postgres",
+								DatabaseName:    "temporal_visibility",
+								ConnectAddr:     connectAddr,
+								ConnectProtocol: "tcp",
+							},
+							PasswordSecretRef: v1beta1.SecretKeyReference{
+								Name: "postgres-password",
+								Key:  "PASSWORD",
+							},
+						},
+					},
+				},
+			}
+			err = client.Resources(namespace).Create(ctx, cluster)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			return context.WithValue(ctx, clusterKey, cluster)
+		}).
+		Assess("Temporal cluster created", AssertClusterReady()).
+		Assess("Can create a temporal cluster namespace", func(ctx context.Context, tt *testing.T, cfg *envconf.Config) context.Context {
+			namespace := GetNamespaceForFeature(ctx)
+
+			// create the temporal cluster client
+			appworker = &v1beta1.TemporalAppWorker{
+				ObjectMeta: metav1.ObjectMeta{Name: "test", Namespace: namespace},
+				Spec: v1beta1.TemporalAppWorkerSpec{
+					Version:  "latest",
+					Replicas: &replicas,
+					Image:    "ktenzer/helloworld-worker",
+					TemporalConnection: &v1beta1.TemporalConnectionSpec{
+						URL:       "test-frontend",
+						Port:      &port,
+						Namespace: namespace,
+					},
+				},
+			}
+			err := cfg.Client().Resources(namespace).Create(ctx, appworker)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			return ctx
+		}).
+		Assess("Temporal app worker created", func(ctx context.Context, tt *testing.T, cfg *envconf.Config) context.Context {
+			err := waitForAppWorker(ctx, cfg, appworker)
+			if err != nil {
+				t.Fatal(err)
+			}
+			return ctx
+
+		}).
+		Assess("Temporal cluster can handle workflows", func(ctx context.Context, tt *testing.T, cfg *envconf.Config) context.Context {
```

Comment:
I think this does nothing with the app worker ?
Can't we start a workflow on the app worker ?